CustomDropDown/View/CustomDropDownView.swift

```diff
@@ -0,0 +1,264 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setConfig() {
+        guard let config = self.presenter?.datasource?.config(identifier: identifier) else { return }
+        
+        self.config = config
+        
+        if T.self == ImageLabelData.self {
+            self.config.dropDownMode = .imageLabel
+        } else if T.self == String.self {
+            self.config.dropDownMode = .label
+        } else {
+            fatalError("not supported item tyoe")
+        }
+    }
+    
+    override func didMoveToSuperview() {
+        super.didMoveToSuperview()
+
+        if let view = self.superview?.viewWithTag(config.dropDownTag) {
+            view.removeFromSuperview()
+        }
+        
+        resetDropDownConstraints()
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        //location is relative to the current view
+        // do something with the touched point
+        super.touchesBegan(touches, with: event)
+        toggleDropDown()
+    }
+    
+    func getDisplayView() {
+        return
+    }
+    
+    // MARK: - DropDown
+    
+    private func setupDropDownDisplayView() {
+        if let view = presenter?.datasource?.overrideDropDownView(identifier: identifier) {
+            dropDownDisplayView = view
+        } else{
+            let tag = config.selectedLabelTag
+            dropDownDisplayView = DropDownDisplayView(tag: tag)
+        }
+    }
+    
+    private func setupDropDown() {
+        dropDown = CustomDropDown<T>(dropDownView: self, tag: config.dropDownTag)
+        dropDown?.translatesAutoresizingMaskIntoConstraints = false
+    }
+    
+    private func resetDropDownConstraints() {
+        guard let dropDown = dropDown else { return }
+        
+        superview?.addSubview(dropDown)
+        superview?.bringSubviewToFront(dropDown)
+        
+        let dropDownLeftRightPadding: UIEdgeInsets = config.dropDownLeftRightPadding
+        
+        dropDown.topAnchor.constraint(equalTo: bottomAnchor, constant: -dropDownLeftRightPadding.top).isActive = true
+        dropDown.leftAnchor.constraint(equalTo: leftAnchor, constant: dropDownLeftRightPadding.left).isActive = true
+        
+        if let dropDownWidth = config.dropDownWidth {
+            dropDown.widthAnchor.constraint(equalToConstant: dropDownWidth).isActive = true
+        } else {
+            dropDown.rightAnchor.constraint(equalTo: rightAnchor, constant: -dropDownLeftRightPadding.right).isActive = true
+        }
+        
+        heightConstraint = dropDown.heightAnchor.constraint(equalToConstant: 0)
+    }
+    
+    private func toggleDropDown() {
+        guard let dropDown = self.dropDown else { return }
+        
+        if isOpen {
+            isOpen = false
+            setHeightConstraint(to: dropDownHeight, maxHeight: dropDown.tableView.contentSize.height)
+            superview?.bringSubviewToFront(dropDown)
+            animateDropDown {
```

Comment:
Can you call animateDropDown method like this? So that no issues are faced in older xcode and going forward good for you! After the update check if the build is passing and request a review!

<img width="625" alt="Screenshot 2020-10-23 at 3 15 34 PM" src="https://user-images.githubusercontent.com/29634986/96988990-b5612700-1542-11eb-82d0-f8536f7bae55.png">